qemu/tests/live_backup_add_bitmap.py

```diff
@@ -0,0 +1,79 @@
+"""
+Bitmap-add test.
+
+Add a bitmap with persistent on/off to drive with image raw/qcow2 and verify
+its existence after system reboot.
+"""
+import logging
+
+from virttest import error_context
+from provider import block_dirty_bitmap
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """Bitmap test.
+
+    1. get target image and drive.
+    2. add bitmap.
+    3. check bitmap existence.
+        a) if existed, check after system_powerdown
+
+    :param test: vt test object
+    :param params: test parameters dictionary
+    :param env: test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    # wait till boot finishes
+    vm.wait_for_login(timeout=int(params.get("login_timeout", 360))).close()
+
+    bitmaps = {}
+    error_context.context("add dirty bitmap")
+    for bitmap in params.get("bitmaps", "").split():
+        bitmap_params = params.object_params(bitmap)
+        bitmaps[bitmap] = bitmap_params
+        target_image = bitmap_params.get("target_image")
+        target_image_params = params.object_params(target_image)
+        block_dirty_bitmap.block_dirty_bitmap_add(vm, bitmap, bitmap_params,
+                                                  target_image,
+                                                  target_image_params)
+
+    # check bitmap existence
+    error_context.context("check bitmap existence", logging.info)
+    bitmap_list = block_dirty_bitmap.get_bitmaps(vm.monitor.info("block"))
+    logging.debug("bitmaps:\n%s", bitmap_list)
+    msgs = []
+    for bitmap, bitmap_params in bitmaps.items():
+        existence = bitmap_params.get("existence") == "yes"
+        if not block_dirty_bitmap.check_bitmap_existence(bitmap_list, bitmap,
+                                                         existence):
+            msg = "bitmap %s %s exists after addition" % (
+                bitmap, "not" if existence else "")
+            msgs.append(msg)
+    if msgs:
+        test.fail("\n".join(msgs))
+
+    # check bitmap existence after shutdown
+    error_context.context("system powerdown", logging.info)
+    vm.monitor.system_powerdown()
```

Comment:
After line 59, the if nothing goes wrong the guest should be off.
So for line 60, you can just use `vm.is_dead()`?

Reply:
`vm.wait_for_shutdown` just wait until guest shuts down, it doesn't send the actual shutdown command I believe.